src/NHibernate/Loader/Loader.cs

```diff
@@ -972,6 +972,34 @@ private void CheckVersion(int i, IEntityPersister persister, object id, object e
 					entry.LockMode = lockMode;
 				}
 			}
+
+			CacheByUniqueKey(i, persister, obj, session);
+		}
+
+		private void CacheByUniqueKey(int i, IEntityPersister persister, object obj, ISessionImplementor session)
+		{
+			// #1226: If it is already loaded and can be loaded from an association with a property ref, make
+			// sure it is also cached by its unique key.
+			var ukName = OwnerAssociationTypes?[i]?.RHSUniqueKeyPropertyName;
```

Comment:
can we unfold `?` operator here? It looks unreadable.

Reply:
Just null propagation all the way. Original code was:
```c#
IAssociationType[] ownerAssociationTypes = OwnerAssociationTypes;
if (ownerAssociationTypes != null && ownerAssociationTypes[i] != null)
{
	string ukName = ownerAssociationTypes[i].RHSUniqueKeyPropertyName;
```

Would you rather have:
```c#
var ownerAssociationTypes = OwnerAssociationTypes;
if (ownerAssociationTypes == null)
	return;
var ukName = ownerAssociationTypes[i]?.RHSUniqueKeyPropertyName;
```